packages/front-end/components/Settings/PaymentInfo.tsx

```diff
@@ -0,0 +1,198 @@
+import { useState, useEffect, useCallback } from "react";
+import { Flex, Text } from "@radix-ui/themes";
+import { CiCreditCard1 } from "react-icons/ci";
+import { Card } from "shared/src/types/subscriptions";
+import { useAuth } from "@/services/auth";
+import { useUser } from "@/services/UserContext";
+import track from "@/services/track";
+import { GBAddCircle } from "../Icons";
+import Callout from "../Radix/Callout";
+import Badge from "../Radix/Badge";
+import MoreMenu from "../Dropdown/MoreMenu";
+import DeleteButton from "../DeleteButton/DeleteButton";
+import Modal from "../Modal";
+import LoadingSpinner from "../LoadingSpinner";
+import CreditCardModal from "./CreditCardModal";
+
+export default function PaymentInfo() {
+  const [cardModal, setCardModal] = useState(false);
+  const [defaultCard, setDefaultCard] = useState<string | undefined>(undefined);
+  const [error, setError] = useState<string | undefined>(undefined);
+  const [cardData, setCardData] = useState<Card[]>([]);
+  const [loadingCards, setLoadingCards] = useState(false);
+  const { subscription } = useUser();
+  const { apiCall } = useAuth();
+
+  const customerId = "cus_Rg3aee6F7wi9EH"; //MKTODO: Fix this so the customer is from the subscription
+
+  const fetchCardData = useCallback(async () => {
+    setLoadingCards(true);
+    try {
+      const res: { cards: Card[] } = await apiCall(
+        `/subscription/payment-methods/${customerId}`,
+        {
+          method: "GET",
+        }
+      );
+      setCardData(res.cards);
+    } catch (e) {
+      setError(e.message);
+    } finally {
+      setLoadingCards(false);
+    }
+  }, [customerId, apiCall, setCardData, setLoadingCards, setError]);
+
+  async function setCardAsDefault() {
+    if (!defaultCard) throw new Error("Must specify card id");
+    try {
+      await apiCall("/subscription/payment-methods/set-default", {
+        method: "POST",
+        body: JSON.stringify({
+          paymentMethodId: defaultCard,
+          customerId,
+        }),
+      });
+      fetchCardData();
```

Comment:
These `fetchCardData()` calls are the expensive way to do it (it refetches the card data from Stripe) but enables us to not have to worry about managing the state and having a `clean` and `dirty` state.

Reply:
Alternatively, we can just call `setCardData` directly and update the `isDefault` field (remove from the old default, set it on the new one).  It seems safe to assume the rest of the data is unaffected and does not need to be refreshed.

Same when detaching a card.  We do probably need to do a full refetch when a new card is added though.

Reply:
I've got this all built out - I am wondering we instead want to disable the "delete" option if it's a default card.

As noted in the meeting, Stripe doesn't require any of the cards on file to be "default". But looking at Orb, if no default card is set, they use the first card in the array of cards as the default. 

Currently I have the logic on our end mirroring that.

But I could disable the delete button for the default card (and add a tooltip) so the user has to explicitly set the default card to avoid potential issues.

Thoughts?

As an aside - do you want to stick with the 3 card limit?

Reply:
Yes, lets do both.  Max of 3 cards and you cannot delete the one marked as default (if you have an active subscription).  Also, when you add your first card, it is always marked as the default (disable/remove the toggle).

I don't want to rely on the "first" card logic.  We should remove all of that from our code.  So some customers today may have no cards marked as "default", which is fine.  Going forward, new customers will always have a default.